CppSamples/EditData/ManageFeaturesFeatureService/ManageFeaturesFeatureService.h

```diff
@@ -14,59 +14,84 @@
 // limitations under the License.
 // [Legal]
 
-#ifndef UPDATE_ATTRIBUTES_FEATURE_SERVICE_H
-#define UPDATE_ATTRIBUTES_FEATURE_SERVICE_H
+#ifndef MANAGEFEATURESFEATURESERVICE_H
+#define MANAGEFEATURESFEATURESERVICE_H
 
 // Qt headers
 #include <QQuickItem>
 
 namespace Esri::ArcGISRuntime
 {
-  class Map;
-  class MapQuickView;
-  class FeatureLayer;
-  class ServiceFeatureTable;
   class ArcGISFeature;
-  class IdentifyLayerResult;
+  class Feature;
+  class FeatureLayer;
   class FeatureQueryResult;
-  class FeatureEditResult;
-}
+  class FeatureTableEditResult;
+  class IdentifyLayerResult;
+  class Map;
+  class MapQuickView;
+  class Point;
+  class ServiceGeodatabase;
+} // namespace Esri::ArcGISRuntime
 
 class QString;
 
-class UpdateAttributesFeatureService : public QQuickItem
+class ManageFeaturesFeatureService : public QQuickItem
 {
   Q_OBJECT
 
+  enum class OperationMode
+  {
+    AddFeatures,
+    DeleteFeatures,
+    UpdateAttributes,
+    UpdateGeometry
+  };
+  Q_ENUM(OperationMode)
+
   Q_PROPERTY(QString featureType READ featureType NOTIFY featureTypeChanged)
+  Q_PROPERTY(OperationMode operationMode READ operationMode WRITE setOperationMode NOTIFY operationModeChanged)
 
 public:
-  explicit UpdateAttributesFeatureService(QQuickItem* parent = nullptr);
-  ~UpdateAttributesFeatureService() override;
+  explicit ManageFeaturesFeatureService(QQuickItem* parent = nullptr);
+  ~ManageFeaturesFeatureService() override;
 
   void componentComplete() override;
   static void init();
+
+  Q_INVOKABLE void deleteSelectedFeature();
   Q_INVOKABLE void updateSelectedFeature(QString fieldVal);
```

Comment:
same const question for this as well

Reply:
Cannot be const beacuse we are updating `m_serviceGeodatabase`

Reply:
apologies for the confusion const ref the parameter
```suggestion
  Q_INVOKABLE void updateSelectedFeature(const QString& fieldVal);
```